Underscore.cs/List/Implementation/Partition.cs

```diff
@@ -142,8 +166,54 @@ public IEnumerable<IEnumerable<T>> Chunk<T>( IList<T> list, Func<T, bool> on )
         /// <param name="list">The list to take the slice from</param>
         /// <param name="start">The start index</param>
         /// <param name="end">The end index</param>
+        /// <param name="allowOverflow">specifies if the slice should cycle on overflow</param>
         /// <returns>slice of the list</returns>
-        public IList<T> Slice<T>( IList<T> list, int start, int end )
+        public IList<T> Slice<T>(IList<T> list, int start, int end, bool allowOverflow)
+        {
+            if (!allowOverflow)
+            {
+                if (start < -list.Count)
+                {
+                    throw new IndexOutOfRangeException(
+                        "start index value must be greater than or equal to -list.Count unless allowOverflow option is used, actual value was " +
+                        start);
+                }
+
+                if (start >= list.Count)
+                {
+                    throw new IndexOutOfRangeException(
+                        "start index value must be less than list.Count unless allowOverflow option is used, actual value was " +
+                        start);
+                }
+
+                if (end < -list.Count)
+                {
+                    throw new IndexOutOfRangeException(
+                        "end index value must be greater than or equal to -list.Count unless allowOverflow option is used, actual value was " +
+                        end);
+                }
+
+
+                if (end >= list.Count)
+                {
+                    throw new IndexOutOfRangeException(
+                        "end index value must be less than list.Count unless allowOverflow option is used, actual value was " +
+                        end);
+                }
+
+                if ((start < 0 && end >= 0) || (start >= 0 && end < 0))
```

Comment:
No it won't, if you have a negative start and a positive end or vice-a-versa then this will raise an exception, which is different than the index out of range exception

There is a test for it so I know it works